File: backend/app/models/vote.py
from database.db import db
from app.models.base_model import BaseModel

class Vote(db.Model, BaseModel):
    __tablename__ = 'vote'
    
    user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=False)
    translation_id = db.Column(db.Integer, db.ForeignKey('translation.id'), nullable=False)
    is_upvote = db.Column(db.Boolean, nullable=False)
    
    user = db.relationship('User', backref=db.backref('votes', lazy=True))
    translation = db.relationship('Translation', backref=db.backref('votes', lazy=True))

File: dashboard/src/models/project/vote.ts
import type { IApiUser } from '../user/user'
import type { IApiTranslation } from './translation'

export interface IApiVote {
  id: number
  created_at: Date
  user_id: number
  user?: IApiUser
  translation_id: number
  translation?: IApiTranslation
  is_upvote: boolean
}
